puppetry/kinectv2/pygame_button.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+"""\
+@file pygame_button.py
+
+@brief Simple button class so this can have basic UI
+
+$LicenseInfo:firstyear=2022&license=viewerlgpl$
+Second Life Viewer Source Code
+Copyright (C) 2022, Linden Research, Inc.
+ 
+This library is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation;
+version 2.1 of the License only.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+Linden Research, Inc., 945 Battery Street, San Francisco, CA  94111  USA
+$/LicenseInfo$
+"""
+
+
+
+import pygame
+
+LEFT_CLICK = 0
+
+BTN_TEXT_SIZE = 28
+BTN_COLOR = (64,64,64)
+BTN_HOVER_COLOR = (1,160,160)
+BTN_TEXT_COLOR = (255,255,255)
+
+#Pygame Colors
+# pgBlack,pgWhite,pgDarkGrey,pgLightGrey = (0,0,0),(255,255,255),(150,150,150),(211,211,211)
+# pgRed,pgDarkRed,pgLightRed,pgReallyLightRed = (255,0,0),(150,0,0),(255,100,100),(255,200,200)
+# pgGreen,pgDarkGreen,pgLightGreen,pgReallyLightGreen = (0,255,0),(0,150,0),(100,255,100),(200,255,200)
+# pgBlue,pgDarkBlue,pgLightBlue,pgReallyLightBlue = (0,0,255),(0,0,150),(100,100,255),(200,200,255)
+# pgYellow,pgDarkYellow,pgGold = (255,255,0),(200,200,0),(212,175,55)
+
+
+#Display images
+# def imageDisplay(Image_name,Width,Height,x,y):
+#     Image = pygame.image.load(f"{Image_name}")
+#     Image = pygame.transform.scale(Image,(Width,Height))
+#     ImageRect = Image.get_rect()
+#     ImageRect.center = (x,y)
+#     Window.blit(Image,ImageRect)
+#     return ImageRect.center,Image
+
+# For logging, set log_function to a function that expects a string
+import puppetry
+
+PYGLogger = puppetry.log
+
+def log_msg(message):

```

Comment:
This function is unused. (Except in a commented-out section)

Reply:
Re-wrote that logging hook a bit and actually use it.   The logging hook really could be removed but was useful when developing and debugging.   It would be nice to have in the future if this ever evolved into more fancy buttons